storeQuestionService/store-q-service.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
+const Question = require('./store-q-model');
+
+const app = express();
+const port = 8004;
+
+// Middleware to parse JSON in request body
+app.use(bodyParser.json());
+
+// Connect to MongoDB
+const mongoUri = process.env.MONGODB_URI || 'mongodb://localhost:27017/storedquestion';
+mongoose.connect(mongoUri);
+
+// Function to validate required fields in the request body
+function validateRequiredFields(req, requiredFields) {
+    for (const field of requiredFields) {
+        if (!(field in req.body)) {
+            throw new Error(`Missing required field: ${field}`);
+        }
+    }
+}
+
+app.post('/addquestion', async (req, res) => {
```

Comment:
- No es necesario poner "add" en la ruta de un endpoint, la propia naturaleza del endpoint (si es GET, POST, DELETE...) ya te indica la operación que se va a realizar en dicha ruta.
- Añadiría el `/history` delante para representar mejor el servicio. Siguiendo este convenio todas las rutas de este servicio serían iguales y las del otro servicio que estáis haciendo serían `/history/matches` o algo así

```suggestion
app.post('/history/questions', async (req, res) => {
```